app/scripts/modules/core/src/artifact/ExpectedArtifactSelectorViewController.ts

```diff
@@ -27,7 +27,10 @@ export class ExpectedArtifactSelectorViewController {
     } else {
       const artifact = ExpectedArtifactService.artifactFromExpected(expectedArtifact);
       const allAccounts = this.delegate.getExpectedArtifactAccounts();
-      this.accountsForArtifact = allAccounts.filter(a => a.types.includes(artifact.type));
+      this.accountsForArtifact =
+        artifact.type === 'helm/chart'
+          ? allAccounts.filter(a => a.types.includes(artifact.type) && a.name === artifact.reference)
```

Comment:
I am a little confused as to why the set of accounts must be limited to those matching the artifact's reference.

Artifacts already have an artifactAccount field (https://github.com/spinnaker/kork/blob/master/kork-artifacts/src/main/java/com/netflix/spinnaker/kork/artifacts/model/Artifact.java#L46)

Does that field better suit this purpose?

Reply:
Thanks for your reply @sbwsg. I have rewritten code according to IArtifact interface.